src/iosrv/srv.c

```diff
@@ -384,7 +385,14 @@ dss_srv_handler(void *arg)
 	/** set affinity */
 	rc = hwloc_set_cpubind(dss_topo, dx->dx_cpuset, HWLOC_CPUBIND_THREAD);
 	if (rc) {
-		D_ERROR("failed to set affinity: %d\n", errno);
+		D_ERROR("failed to set cpu affinity: %d\n", errno);
+		goto signal;
+	}
+
+	rc = hwloc_set_membind(dss_topo, dx->dx_cpuset, HWLOC_MEMBIND_BIND,
```

Comment:
I think that HWLOC_CPUBIND_THREAD needs to be one of 

> enum hwloc_membind_flags_t { HWLOC_MEMBIND_PROCESS, HWLOC_MEMBIND_THREAD, HWLOC_MEMBIND_STRICT, HWLOC_MEMBIND_MIGRATE, HWLOC_MEMBIND_NOCPUBIND }.

  Given that, I think you want HWLOC_MEMBIND_THREAD.